src/core/func-core-tools.ts

```diff
@@ -0,0 +1,217 @@
+import os from 'os';
+import fs from 'fs';
+import path from 'path';
+import process from 'process';
+import { exec } from 'child_process';
+import { promisify } from 'util';
+import fetch from 'node-fetch';
+import unzipper from 'unzipper';
+import cliProgress from 'cli-progress';
+import { logger } from "./utils/logger";
+
+const RELEASES_FEED_URL = 'https://functionscdn.azureedge.net/public/cli-feed-v4.json';
+const DEFAULT_FUNC_BINARY = 'func';
+const VERSION_FILE = '.release-version';
+const LOCAL_CORE_TOOLS_FOLDER = '.swa/core-tools';
```

Comment:
Can we store the `func` binary in the node_modules (or similar)? I'd like us to avoid maintaining a new config folder.

Also, users may accidentally commit this binary to their git history.

Reply:
It's located in the user folder at `~/.swa/core-tools`, so it's shared among projects 😉 
Also no risk of commiting it by mistake there.